src/lib/api/projects.ts

```diff
@@ -51,33 +52,43 @@ export async function list(
     endpoint.searchParams.set(k, String(v));
   }
 
-  const res = await fetch(endpoint, { credentials: "include" }).catch((e) => {
-    error(500, { message: e });
-  });
+  const resp = await fetch(endpoint, { credentials: "include" }).catch(
+    console.error,
+  );
+
+  if (!resp) {
+    throw new Error("API error");
+  }
 
-  if (isErrorCode(res.status)) {
-    error(res.status, {
-      message: res.statusText,
-    });
+  if (isErrorCode(resp.status)) {
+    throw new Error(resp.statusText);
```

Comment:
Just to be clear, we don't want to be invoking SvelteKit's `error` handler here? Is this something that should only be called from a page/layout route when we're actually loading the data by calling this function, or is SK gonna handle that error logic for us?

Reply:
We don't. The API is giving us back useful error information, and invoking any error -- SvelteKit's or just `new Error` -- stops us from using that. This is what I'm hoping to solve in #502.